ovsdb/client.go

```diff
@@ -229,6 +277,45 @@ func (c *Client) listen() {
 	}
 }
 
+// echoLoop starts a loop that sends echo RPCs at the interval defined by d.
+func (c *Client) echoLoop(ctx context.Context, d time.Duration) {
+	t := time.NewTicker(d)
+	defer t.Stop()
+
+	for {
+		// Prioritize closed context over a tick on the channel.
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			// Then wait for closed context or a tick to send a new RPC.
+			select {
+			case <-ctx.Done():
+				return
+			case <-t.C:
+			}
+		}
+
+		// For the time being, we will track metrics about the number of successes
+		// and failures while sending echo RPCs.
+		// TODO(mdlayher): improve error handling for echo loop.
+		if err := c.Echo(ctx); err != nil {
+			if isClosedNetwork(err) {
+				// Our socket was closed, which means the context should be canceled
+				// and we should terminate on the next loop.  No need to increment
+				// errors counter.
+				continue
+			}
+
+			// Count other errors as failures.
+			atomic.AddInt64(c.echoFail, 1)
```

Comment:
I like the counter here -- it'd be awesome to be able to reconnect eventually, but this feels like a great first step.

Reply:
Is this `c.echoFail` shared resource?  Do we have multiple goroutines to increment this error counter?  Want to know the reason for the atomicity.

Reply:
Since the Stats method could be called at any time to fetch the value of these counters, as they are incremented by this goroutine, I've used atomics for them.